In [ ]:
pip install apyori

  Created wheel for apyori: filename=apyori-1.1.2-cp36-none-any.whl size=5975 sha256=d88c6059a53463125a7a225ba7c8acfa2f5692c4c960ca6ce51ceda2380eb8bd
  Stored in directory: /root/.cache/pip/wheels/5d/92/bb/474bbadbc8c0062b9eb168f69982a0443263f8ab1711a8cad0
Successfully built apyori


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [ ]:
df=pd.read_csv('Market_Basket_Optimisation.csv',header=None)
transactions=[]
for i in range(len(df.count(axis=1))):
  transactions.append([str(df.values[i,j]) for j in range(len(df.count(axis=0)))])
#before appending the list wwe must change the format to string to make sure
#that the apyori model gets its input in expected string format

In [ ]:
transactions[0]

['shrimp',
 'almonds',
 'avocado',
 'vegetables mix',
 'green grapes',
 'whole weat flour',
 'yams',
 'cottage cheese',
 'energy drink',
 'tomato juice',
 'low fat yogurt',
 'green tea',
 'honey',
 'salad',
 'mineral water',
 'salmon',
 'antioxydant juice',
 'frozen smoothie',
 'spinach',
 'olive oil']

In [ ]:
df.count(axis=0)

0     7501
1     5747
2     4389
3     3345
4     2529
5     1864
6     1369
7      981
8      654
9      395
10     256
11     154
12      87
13      47
14      25
15       8
16       4
17       4
18       3
19       1
dtype: int64



*   **minimum_support:** we chose the minimum support to be chose such as a product that appears 3 times a day,7 days a week & 7501 trasnsactions/week
*   **minimum_confidence:** starting from 0.8 to 0.2


*   **minimum_lift**: good use to take 3
*   **min_length:** for best offer rules

*   **max_length:** for best offer rules









In [ ]:
from apyori import apriori
rules=apriori(transactions,min_support=0.003,min_confidence=0.2,min_lift=3,min_length=2,max_length=2)

In [ ]:
results=list(rules)
results

[RelationRecord(items=frozenset({'chicken', 'light cream'}), support=0.004532728969470737, ordered_statistics=[OrderedStatistic(items_base=frozenset({'light cream'}), items_add=frozenset({'chicken'}), confidence=0.29059829059829057, lift=4.84395061728395)]),
 RelationRecord(items=frozenset({'mushroom cream sauce', 'escalope'}), support=0.005732568990801226, ordered_statistics=[OrderedStatistic(items_base=frozenset({'mushroom cream sauce'}), items_add=frozenset({'escalope'}), confidence=0.3006993006993007, lift=3.790832696715049)]),
 RelationRecord(items=frozenset({'pasta', 'escalope'}), support=0.005865884548726837, ordered_statistics=[OrderedStatistic(items_base=frozenset({'pasta'}), items_add=frozenset({'escalope'}), confidence=0.3728813559322034, lift=4.700811850163794)]),
 RelationRecord(items=frozenset({'honey', 'fromage blanc'}), support=0.003332888948140248, ordered_statistics=[OrderedStatistic(items_base=frozenset({'fromage blanc'}), items_add=frozenset({'honey'}), confidence=0

**Putting into pandas dataframe**

In [ ]:
lhs=[tuple(result[2][0][0])[0] for result in results]
rhs=[tuple(result[2][0][1])[0] for result in results]
support=[result[1] for result in results]
confidence=[result[2][0][2] for result in results]
lift=[result[2][0][3] for result in results]
a=list(zip(lhs,rhs,support,confidence,lift))
df=pd.DataFrame(a,columns=['LHS','RHS','Support','Confidence','Lift'])

In [ ]:
df.sort_values(by='Lift',ascending=False,axis=0)
#SORTING THE DATAFRAME IN DESCENDING ORDER OF LIFT VALUES

LHS          RHS   Support  Confidence      Lift
3         fromage blanc        honey  0.003333    0.245098  5.164271
0           light cream      chicken  0.004533    0.290598  4.843951
2                 pasta     escalope  0.005866    0.372881  4.700812
8                 pasta       shrimp  0.005066    0.322034  4.506672
7     whole wheat pasta    olive oil  0.007999    0.271493  4.122410
5          tomato sauce  ground beef  0.005333    0.377358  3.840659
1  mushroom cream sauce     escalope  0.005733    0.300699  3.790833
4         herb & pepper  ground beef  0.015998    0.323450  3.291994
6           light cream    olive oil  0.003200    0.205128  3.114710